alf/layers.py

```diff
@@ -956,6 +956,161 @@ def bias(self):
         return self._conv2d.bias
 
 
+@alf.configurable
+class Conv2DBatchEnsemble(Conv2D):
+    r"""The BatchEnsemble for 2D Conv layer.
+
+    BatchEnsemble is proposed in `Wen et. al. BatchEnsemble: An Alternative Approach
+    to Efficient Ensemble and Lifelong Learning <https://arxiv.org/abs/2002.06715>`_
+
+    In a nutshell, a tuple of vector :math:`(r_k, s_k)` is maintained for ensemble
+    member k in addition to the original conv2d kernel W. For input x, the
+    result for ensemble member k is calculated as :math:`(W \circ (s_k r_k^T)) * x`.
```

Comment:
For conv2d, W is rank-3 tensor. So this is not quite right.

Reply:
Corrected.